zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/identity/AuthorizationCheckBehavior.java

```diff
@@ -307,119 +279,50 @@ public Set<String> getDirectAuthorizedResourceIdentifiers(
+                                  .stream())
+                      .forEach(stream);
                 });
-    return Stream.concat(groupAuthorizedResourceIdentifiers, roleAuthorizedResourceIdentifiers);
-  }
-
-  private Stream<String> getMappingsAuthorizedResourceIdentifiers(
-      final AuthorizationRequest request, final List<PersistedMapping> persistedMappings) {
-    return persistedMappings.stream()
-        .mapMulti(
-            (mapping, stream) -> {
-              getAuthorizedResourceIdentifiersForOwners(
-                      AuthorizationOwnerType.MAPPING,
-                      List.of(mapping.getMappingId()),
-                      request.getResourceType(),
-                      request.getPermissionType())
-                  .forEach(stream);
-              getGroupsAuthorizedResourceIdentifiers(
-                      EntityType.MAPPING,
-                      mapping.getMappingId(),
-                      request.getResourceType(),
-                      request.getPermissionType())
-                  .forEach(stream);
-              getAuthorizedResourceIdentifiersForOwners(
-                      AuthorizationOwnerType.ROLE,
-                      membershipState.getMemberships(
-                          EntityType.MAPPING, mapping.getMappingId(), RelationType.ROLE),
-                      request.getResourceType(),
-                      request.getPermissionType())
-                  .forEach(stream);
-            });
-  }
-
-  private Stream<String> getAuthorizedResourceIdentifiersForOwners(
-      final AuthorizationOwnerType ownerType,
-      final List<String> ownerIds,
-      final AuthorizationResourceType resourceType,
-      final PermissionType permissionType) {
-    return ownerIds.stream()
-        .flatMap(
-            ownerId ->
-                authorizationState
-                    .getResourceIdentifiers(ownerType, ownerId, resourceType, permissionType)
-                    .stream());
-  }
-
-  /**
-   * Checks the resource identifiers of the request against the authorized resource identifiers of
-   * the entity.
-   *
-   * @param request the authorization request to check authorization for
-   * @param authorizedResourceIdentifiers the authorized resource identifiers of the entity
-   * @return an {@link Either} containing a {@link Rejection} if there is no match or {@link Void}
-   *     if there is.
-   */
-  private Either<Rejection, Void> checkResourceIdentifiers(
-      final AuthorizationRequest request, final Stream<String> authorizedResourceIdentifiers) {
-    final var isAuthorized =
-        authorizedResourceIdentifiers.anyMatch(
-            resourceId -> request.getResourceIds().contains(resourceId));
-    if (isAuthorized) {
-      return Either.right(null);
-    } else {
-      return Either.left(
-          new Rejection(RejectionType.FORBIDDEN, request.getForbiddenErrorMessage()));
-    }
+    return Stream.concat(direct, Stream.concat(viaRole, viaGroups));
```

Comment:
🔧 As per the docs, instead of nesting here we could also use the `Stream.of()` method and flatMap the result:

```
return Stream.of(s1, s2, s3, s4).flatMap(s -> s);
```

Reply:
Hm, I find this version a bit easier to read :shrug: